test/integration/crud/find_cursor_methods.test.js

```diff
@@ -1,259 +1,239 @@
-        this.defer(() => cursor.close());
+    it('should clone a find cursor', function (done) {
+      const coll = client.db().collection('abstract_cursor');
+      const cursor = coll.find({});
+      this.defer(() => cursor.close());
 
-        cursor.toArray((err, docs) => {
+      cursor.toArray((err, docs) => {
+        expect(err).to.not.exist;
+        expect(docs).to.have.length(6);
+        expect(cursor).property('closed').to.be.true;
+
+        const clonedCursor = cursor.clone();
+        this.defer(() => clonedCursor.close());
+
+        clonedCursor.toArray((err, docs) => {
           expect(err).to.not.exist;
           expect(docs).to.have.length(6);
-          expect(cursor).property('closed').to.be.true;
+          expect(clonedCursor).property('closed').to.be.true;
+          done();
+        });
+      });
+    });
 
-          const clonedCursor = cursor.clone();
-          this.defer(() => clonedCursor.close());
+    it('should clone an aggregate cursor', function (done) {
+      const coll = client.db().collection('abstract_cursor');
+      const cursor = coll.aggregate([{ $match: {} }]);
+      this.defer(() => cursor.close());
 
-          clonedCursor.toArray((err, docs) => {
-            expect(err).to.not.exist;
-            expect(docs).to.have.length(6);
-            expect(clonedCursor).property('closed').to.be.true;
-            done();
-          });
-        });
-      })
-    );
+      cursor.toArray((err, docs) => {
+        expect(err).to.not.exist;
+        expect(docs).to.have.length(6);
+        expect(cursor).property('closed').to.be.true;
 
-    it(
-      'should clone an aggregate cursor',
-      withClientV2(function (client, done) {
-        const coll = client.db().collection('abstract_cursor');
-        const cursor = coll.aggregate([{ $match: {} }]);
-        this.defer(() => cursor.close());
+        const clonedCursor = cursor.clone();
+        this.defer(() => clonedCursor.close());
 
-        cursor.toArray((err, docs) => {
+        clonedCursor.toArray((err, docs) => {
           expect(err).to.not.exist;
           expect(docs).to.have.length(6);
-          expect(cursor).property('closed').to.be.true;
-
-          const clonedCursor = cursor.clone();
-          this.defer(() => clonedCursor.close());
-
-          clonedCursor.toArray((err, docs) => {
-            expect(err).to.not.exist;
-            expect(docs).to.have.length(6);
-            expect(clonedCursor).property('closed').to.be.true;
-            done();
-          });
+          expect(clonedCursor).property('closed').to.be.true;
+          done();
         });
-      })
-    );
+      });
+    });
   });
 
-  context('#rewind', function () {
-    it(
-      'should rewind a cursor',
-      withClientV2(function (client, done) {
-        const coll = client.db().collection('abstract_cursor');
-        const cursor = coll.find({});
-        this.defer(() => cursor.close());
+  context.skip('#rewind', function () {
```

Comment:
Is this test intentionally skipped? If it is, should we have a jira ticket linked to it?

Reply:
Unintentional rewind should be working again (this was the strangeness with not awaiting endSession)